types/memoize-one/index.d.ts

```diff
@@ -1,12 +1,9 @@
 // Type definitions for memoize-one 3.1
 // Project: https://github.com/alexreardon/memoize-one#readme
-// Definitions by: Karol Majewski <https://github.com/karol-majewski>
+// Definitions by: Karol Majewski <https://github.com/karol-majewski>, Frank Li <https://github.com/franklixuefei>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-export = memoizeOne;
+declare function memoizeOne<T extends (...args: any[]) => any>(resultFn: T, isEqual?: EqualityFn): T;
```

Comment:
Personally, I'm very much in favor of this change. However, the current way of doing it is how the [`dts-gen` templates](https://github.com/Microsoft/dts-gen/tree/master/templates) suggest:

```ts
/*~ If you want to expose types from your module as well, you can
 *~ place them in this block. Often you will want to describe the
 *~ shape of the return type of the function; that type should
 *~ be declared in here, as this example shows.
 */
declare namespace MyFunction {
    export interface LengthReturnType {
        width: number;
        height: number;
    }
    export interface NamedReturnType {
        firstName: string;
        lastName: string;
    }

    /*~ If the module also has properties, declare them here. For example,
     *~ this declaration says that this code is legal:
     *~   import f = require('myFuncLibrary');
     *~   console.log(f.defaultName);
     */
    export const defaultName: string;
    export let defaultLength: number;
}
```

Or for single classes:

```ts
/*~ If you want to expose types from your module as well, you can
 *~ place them in this block.
 */
declare namespace MyClass {
    export interface MyClassMethodOptions {
        width?: number;
        height?: number;
    }
}
```

@andy-ms — is the above a very strong recommendation, or rather just one way of doing this?

Reply:
The two styles actually have a different meaning -- `export =` would mean that `require("memoize-one")` is itself a function, and `export default` means that `require("memoize-one")` is an object with a `default` property. Since the latter seems to be the case, this PR is correct.